### ISSUE TITLE
haskoin-core-0.3.1 does not compile

### ISSUE THREAD

**Issue by peti** [jprupp/haskoin-core#286]
We cannot build haskoin-core-0.3.1 in openSUSE Linux because of the following error:

```
[  149s] + ./Setup configure -v --prefix=/usr --libdir=/usr/lib64 --docdir=/usr/share/doc/packages/ghc-haskoin-core '--libsubdir=$compiler/$pkgkey' '--datasubdir=$pkgid' --ghc -p --enable-shared --htmldir=/usr/share/doc/ghc-8.0.1/html/libraries/haskoin-core-0.3.1 --global
[  149s] Configuring haskoin-core-0.3.1...
[  149s] Dependency QuickCheck >=2.6 && <2.9: using QuickCheck-2.8.2
[  149s] Dependency aeson >=0.7 && <0.12: using aeson-0.11.2.1
[  149s] Dependency base >=4.8 && <5: using base-4.9.0.0
[  149s] Dependency base16-bytestring ==0.1.*: using base16-bytestring-0.1.1.6
[  149s] Dependency binary >=0.7 && <0.9: using binary-0.8.3.0
[  149s] Dependency byteable ==0.1.*: using byteable-0.1.1
[  149s] Dependency bytestring ==0.10.*: using bytestring-0.10.8.1
[  149s] Dependency conduit ==1.2.*: using conduit-1.2.7
[  149s] Dependency containers ==0.5.*: using containers-0.5.7.1
[  149s] Dependency cryptohash ==0.11.*: using cryptohash-0.11.9
[  149s] Dependency deepseq ==1.4.*: using deepseq-1.4.2.0
[  149s] Dependency either >=4.3 && <4.5: using either-4.4.1.1
[  149s] Dependency entropy ==0.3.*: using entropy-0.3.7
[  149s] Dependency largeword >=1.2.4 && <1.3: using largeword-1.2.5
[  149s] Dependency mtl ==2.2.*: using mtl-2.2.1
[  149s] Dependency murmur3 ==1.0.*: using murmur3-1.0.3
[  149s] Dependency network ==2.6.*: using network-2.6.3.1
[  149s] Dependency pbkdf ==1.1.*: using pbkdf-1.1.1.1
[  149s] Dependency secp256k1 ==0.4.*: using secp256k1-0.4.6
[  149s] Dependency split ==0.2.*: using split-0.2.3.1
[  149s] Dependency string-conversions ==0.4.*: using string-conversions-0.4
[  149s] Dependency text >=0.11 && <1.3: using text-1.2.2.1
[  149s] Dependency time >=1.4 && <1.7: using time-1.6.0.1
[  149s] Dependency vector >=0.10 && <0.12: using vector-0.11.0.0
[  150s] Using Cabal-1.24.0.0 compiled by ghc-8.0
[  150s] Using compiler: ghc-8.0.1
[  150s] Using install prefix: /usr
[  150s] Binaries installed in: /usr/bin
[  150s] Libraries installed in:
[  150s] /usr/lib64/ghc-8.0.1/haskoin-core-0.3.1-5J05DsNzffr3YyBTF0fWsw
[  150s] Private binaries installed in: /usr/libexec
[  150s] Data files installed in: /usr/share/haskoin-core-0.3.1
[  150s] Documentation installed in: /usr/share/doc/packages/ghc-haskoin-core
[  150s] Configuration files installed in: /usr/etc
[  150s] No alex found
[  150s] Using ar found on system at: /usr/bin/ar
[  150s] No c2hs found
[  150s] No cpphs found
[  150s] Using gcc version 6 found on system at: /usr/bin/gcc
[  150s] Using ghc version 8.0.1 found on system at: /usr/bin/ghc
[  150s] Using ghc-pkg version 8.0.1 found on system at: /usr/bin/ghc-pkg
[  150s] No ghcjs found
[  150s] No ghcjs-pkg found
[  150s] No greencard found
[  150s] Using haddock version 2.17.2 found on system at: /usr/bin/haddock
[  150s] No happy found
[  150s] Using haskell-suite found on system at: haskell-suite-dummy-location
[  150s] Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
[  150s] No hmake found
[  150s] Using hpc version 0.67 found on system at: /usr/bin/hpc
[  150s] Using hsc2hs version 0.68 found on system at: /usr/bin/hsc2hs
[  150s] No hscolour found
[  150s] No jhc found
[  150s] Using ld found on system at: /usr/bin/ld
[  150s] No lhc found
[  150s] No lhc-pkg found
[  150s] Using pkg-config version 0.29.1 found on system at: /usr/bin/pkg-config
[  150s] Using strip version 2.26 found on system at: /usr/bin/strip
[  150s] Using tar found on system at: /bin/tar
[  150s] No uhc found
[  150s] Setup: configure script not found.
[  150s] error: Bad exit status from /var/tmp/rpm-tmp.BR8BYn (%build)
```

This looks like an incomplete release archive?

**Comment by peti** [comment]
The problem is that your `Setup.hs` file is wrong, i.e. it tries to run an autoconf build which your package doesn't actually use. That file should look as follows:

```
import Distribution.Simple
main = defaultMain
```

**Comment by jprupp** [comment]
Thank you very much for finding this solution. I will upload a new version with this change that you have suggested.

**Comment by jprupp** [comment]
Fixed in commit bdbfacb581b67531cfc32be61db1c97d82e2e194.
